custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -385,8 +380,20 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
             vm.onSelectYear({'id': vm.selectedYear});
             vm.selectedFormat = 'xlsx';
         }
+
+        vm.adjustSelectedLevelForNoViewByFilter();
+
     };
 
+    /**
+     * To adjust selectedLevel for the reports that do not have viewBy filter. These
+     * reports end up having selectedLevel set by viewBy filter in the last report selected.
+     */
+    vm.adjustSelectedLevelForNoViewByFilter = function () {
+        if (!vm.showViewBy() || vm.isTakeHomeRationReportSelected()) {
```

Comment:
should `vm.isTakeHomeRationReportSelected()` be a component of `vm.showViewBy()`

Reply:
yeah it can be there. put it there